lib/remote/endpoint.hpp

```diff
@@ -61,6 +70,9 @@ class Endpoint final : public ObjectImpl<Endpoint>
 	mutable RingBuffer m_MessagesReceived{60};
 	mutable RingBuffer m_BytesSent{60};
 	mutable RingBuffer m_BytesReceived{60};
+
+	mutable std::shared_mutex m_MessageCountersMutex;
+	std::unordered_map<Object::Ptr, std::pair<String, Atomic<uintmax_t, std::memory_order_relaxed>>> m_MessageCounters;
```

Comment:
Isn't this supposed to represent RPC methods that are of type `ApiFunction::Ptr`? So why not use that type directly here instead of the generic `Object::Ptr`? Doing it that way, you could set the name of the function when registering it and discard this extra `std::pair()`.
https://github.com/Icinga/icinga2/blob/061338156c71457397f4127a6c9479cdddc9f0e9/lib/remote/apifunction.hpp#L50-L54